thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/models/AnomalyResultModel.js

```diff
@@ -30,6 +30,22 @@ function AnomalyResultModel() {
 }
 
 AnomalyResultModel.prototype = {
+  hasSameParams(params) {
+    if (!params.anomaliesSearchMode || !this.anomaliesSearchMode) {
+      return false;
+    }
+    const paramKeys = Object.keys(params);
+    for (let index in paramKeys){
```

Comment:
I'd use `Array#every` in this case, it's more declarative than imperatively going through with a `for...in` since you're trying to check that each element meets a condition. Also allows for easier to read constructs, less lines and more maintainable. I would also add a comment so future you knows exactly what the intention of this code is and the "why".

It looks like you are trying to check, in the positive, that?
- each `param` in `params` is a *tab* or a *rand*
- each `param` is `HASH_PARAMS.isSame` as `this[param]`


If so, you can do:

````javascript
// **You can modify this comment, not fully aware of the context here
// Check that each param is either a tab or a rand
// and is the same as the HASH_PARAMS
return Object.keys(params).every((param) => {
  const isParamTabOrRand = ['tab', 'rand'].includes(param);
  const isParamSameAsHash = HASH_PARAMS.isSame(param, params[param], this[param]);

  return isParamTabOrRand && isParamSameAsHash;
});